src/praisonai-agents/praisonaiagents/agents/agents.py

```
@@ -501,14 +501,40 @@ def _process_task_result(agents_instance, context, agent_output):
             try:
                 parsed = json.loads(cleaned)
-                pyd_obj = task.output_pydantic(**parsed)
+                if hasattr(task.output_pydantic, "model_validate"):
+                    pyd_obj = task.output_pydantic.model_validate(parsed)
+                else:
+                    pyd_obj = task.output_pydantic(**parsed)
                 task_output.pydantic = pyd_obj
                 task_output.output_format = "Pydantic"
-            except Exception:
-                logger.warning(f"Warning: Could not parse output of task {task_id} as Pydantic Model")
+            except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)